cli/util/eta.go

```diff
@@ -0,0 +1,74 @@
+package cliutil
+
+import (
+	"fmt"
+	"time"
+)
+
+// ETA implements a very simple eta calculator based on the number of remaining items. It does not
+// require knowing the work size in advance and is therefore suitable for streaming workloads and
+// also does not require that consecutive updates have a monotonically decreasing remaining value.
```

Comment:
Off topic: I'm curious why you chose sliding window over exponentially weighted moving average. Sliding window is probably _better_, I'm just usually too lazy for that and was wondering what your reasoning was.